File: Algorithms/Python Algorithms/Sort Algorithms/bubble_sort.py
#Bubble sort is a sorting algorithm that compares
#two adjacent elements and swaps them if they are
#not in the intended order.

def check_sort(array):
    true_sort = 0
    for i in range(0, len(array) - 1):
        if array[i] < array[i + 1]:
            true_sort += 1
    if true_sort == len(array) - 1:
        return True
    else:
        return False

def bubble_sort(array):
    current_element = 0
    while not check_sort(array):
        for i in range(0, len(array) - 1):
            if array[i] > array[i + 1]:
                current_element = array[i]
                array[i] = array[i + 1]
                array[i + 1] = current_element

    return array


my_list = [-2, 0, 11, -9, 45, 102, -12]
print(bubble_sort(my_list))

File: Algorithms/C++ Algorithms/Sort Algorithms/bubble_sort.cpp
#include <iostream>
using namespace std;

void bubbleSort(int array[], int size);
bool checkSort(int array[], int size);
void printArray(int array[], int size);

int main(){

  int myArray[] = {-2, 0, 11, -9, 45};
  int n = sizeof(myArray) / sizeof(myArray[0]);
  cout << "Unsorted array: ";
  printArray(myArray, n);
  bubbleSort(myArray, n);
  cout << "Sorted array: ";
  printArray(myArray, n);
  cout << endl;

  return 0;
}

void bubbleSort(int array[], int size){

  int currentElement = 0;

  while (checkSort(array, size) == 0){
    for (int i = 0; i < size - 1; ++i){
      if (array[i] > array[i + 1]){
        currentElement = array[i];
        array[i] = array[i + 1];
        array[i + 1] = currentElement;
      }
    }
  }

}

bool checkSort(int array[], int size){
  int trueSort = 0;
  for(int i = 0; i < size - 1; ++i){
    if(array[i] < array[i + 1]){
      trueSort += 1;
    }
  }

  if(trueSort == size - 1){
    return true;
  }else{return false;}

}

void printArray(int array[], int size){
  cout << "[";
  for(int i = 0; i < size; ++i){
    if (i == size - 1){
      cout << array[i];
    }else {cout << array[i] << ", ";}
  }

  cout << "]" <<endl;
}
